test_pull_requests

```diff
@@ -505,34 +660,27 @@ module Hub
     def mark_test_success(pull_id, repo, settings)
       $stderr.puts "  Marking SUCCESS for pull request ##{pull_id} in repo '#{repo}'"
       bot_comment = get_comment_with_prefix(pull_id, repo, settings['test_prefix'])
-      build_url = nil
-      extended_tests = nil
-      if bot_comment['body'] =~ /^#{settings['test_prefix']} Running \(([^\)]+)\)( \(Extended Tests: [^\)]+\).*)?/
-        build_url = $1
-        extended_tests = $2
-      else
-        raise "'#{settings['name']}' pull request not found"
-      end
-      comment = "#{settings['test_prefix']} SUCCESS (#{build_url})"
-      comment += extended_tests if extended_tests
+      fields = extract_bot_comment_fields(bot_comment['body'], settings && settings['test_prefix'], settings && settings['name'])
```

Comment:
Actually no, it was a bad solution. @stevekuznetsov it's OK to pass around a reference to a config settings object instead of just the bits of data that you need from it. In this instance, since the absence or presence of that object informs how the method will behave, and the method is used throughout the program, passing around the object reference is I think the correct thing to do.